src/infra/vocabulary/sparql/sparqlGateway.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable max-lines-per-function */
+import type {
+  RetrieveVocabularyResult,
+  RetrieveVocabularyError,
+  VocabularyPort,
+  VocabularyType
+} from '@/domain/vocabulary/port'
+import { fetchWithSparql, serializeFetchResponse } from '@/infra/shared/sparql.util'
+import * as TE from 'fp-ts/TaskEither'
+import * as A from 'fp-ts/Array'
+import { pipe } from 'fp-ts/lib/function'
+import type { SparqlResult } from './dto'
+import type { VocabularyElement } from '@/domain/vocabulary/entity'
+
+export const sparqlGateway: VocabularyPort = {
+  retrieveVocabulary: (
+    type: VocabularyType,
+    language: string,
+    limit: number,
+    offset: number
+  ): TE.TaskEither<RetrieveVocabularyError, RetrieveVocabularyResult> => {
+    const query = `
+      PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+	    PREFIX thesaurus: <https://ontology.okp4.space/thesaurus/>
+      SELECT ?term ?prefLabel
+      WHERE {
+        ?term a skos:Concept .
+        ?term skos:inScheme thesaurus:${type} .
```

Comment:
Injecting the type value directly into the query is a weak approach because it assumes the symbol is an accurate representation (this is a practice that many frontend developers tend to do, mainly because TypeScript encourages it). This could silently generate incorrect queries, and there's no mechanism in place for type checking or validation.

I would suggest to explicitly convert the type to its corresponding representation, I mean its URI. This is important from a functional perspective, because not all vocabularies are necessarily part of our namespace.

Reply:
@ccamel 
I made the modifications in the union types, now the URI is the argument of the gateway.